File: week-2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text:");

    int words = 1;
    int letters = 0;
    int sentence = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        text[i] = toupper(text[i]);
        int index = text[i] - 'A';
        if (index >= 0 && index < 26)
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence++;
        }
    }

    float L = ((float)letters / words) * 100;
    float S = ((float)sentence / words) * 100;

    int grade = round(0.0588 * L - 0.296 * S - 15.8);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

    return 0;
}

File: week-6/readability.py
text = input("Text:")

words = 1
letters = 0
sentence = 0

for i in range(len(text)):
    char = text[i].upper()
    index = ord(char) - ord('A')

    if index >= 0 and index < 26:
        letters += 1
    elif char == ' ':
        words += 1
    elif char == '.' or char == '!' or char == '?':
        sentence += 1

L = (letters / words) * 100
S = (sentence / words) * 100

grade = round(0.0588 * L - 0.296 * S - 15.8)

if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")